packages/logger/src/services/json-encoder.ts

```diff
@@ -0,0 +1,157 @@
+import { SafeFunction, TypeSpecifier } from './type-specifier'
+
+const INDENTATION_WIDTH = 2n
+
+export interface JsonEncoderOptions {
+    isPretty: boolean
+}
+
+export class JsonEncoder {
+
+  private readonly typeSpecifier = new TypeSpecifier()
+  private readonly isPretty: boolean
+
+  constructor({ isPretty }: JsonEncoderOptions) {
+    this.isPretty = isPretty
+  }
+
+  public encode(data: unknown): string {
+    const depth = 0n
+    return this.encodeRecursively(data, depth)
+  }
+
+  private encodeRecursively(data: unknown, depth: bigint): string {
```

Comment:
Is it really not possible to make use of a switch, instead of all those if's?

Reply:
Not really. If Javascript supported pattern matching and was strongly typed, then yes.